shapeshift/web
# feat: add Pendo plugin

## Description

Add support for Pendo in app, which can be enabled or disabled based on a feature flag.

* This code will always be compiled into the app, which means there will not be a version of the app that does not have the pendo client code.
* The code to initiate a connection with Pendo should/will only run when the feature flag is enabled
* This patches a specific version of the pendo agent to remove specific security vulnerabilities. That makes changing the pendo agent version _very_ difficult.

## Notice




## Pull Request Type


## Issue (if applicable)

There is one, I'll edit when I find it.

## Risk

* The pendo launcher could potentially cause delays in loading the app
* There is some risk of security vulnerabilities exposed due to including the pendo agent code, even if the flag is off
* It's possible the pendo agent could fail in ways that would crash the app

## Testing

1. We must add PENDO specific environment variables to the build process
2. Run the app with PENDO enabled and review the Network tab to verify that requests to pendo are working
3. Run the app with PENDO disabled and review the network tab to verify that requests to pendo are not happening
4. Check the Console tab for any pendo related logs
5. Check the Pendo console to verify that data is arriving

## Screenshots (if applicable)
![image](https://user-images.githubusercontent.com/1958266/178788460-5f4391fd-e021-40fb-bd41-fe0c2d3d1eaa.png)
